### ISSUE TITLE
Check for invalid HTML elements nested inside an <a>

### ISSUE THREAD

**Issue by jackdomleo7** [jackdomleo7/Checka11y.css#16]
Nesting the following inside an `<a>` is invalid:

```html
    <a>
    <audio> (if the controls attribute is present)
    <button>
    <details>
    <embed>
    <iframe>
    <img> (if the usemap attribute is present)
    <input> (if the type attribute is not in the hidden state)
    <keygen>
    <label>
    <menu> (if the type attribute is in the toolbar state)
    <object> (if the usemap attribute is present)
    <select>
    <textarea>
    <video> (if the controls attribute is present)
```

Evidence: https://stackoverflow.com/questions/6393827/can-i-nest-a-button-element-inside-an-a-using-html5/6393863#6393863

**Solution**

Show an error in the above elements are detected nested anywhere inside an `<a>`.

**Comment by wulforr** [comment]
I would like to implement this if no one is already implementing it.

**Comment by jackdomleo7** [comment]
Sounds good! Be my guest. Be sure to pay close attention to the information in brackets too.

**Comment by wulforr** [comment]
Sure I will keep them in mind
